bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -243,15 +246,30 @@ public String toString() {
      */
     @Override
     public void stateChanged(Item item, State oldState, State newState) {
-        setState(function.calculate(members));
+        setStateWithouEvent(function.calculate(members));
+        sendGroupStateChangedEvent(item.getName(), this.state, oldGroupState);
     }
-
+    
     /**
      * @{inheritDoc
      */
     @Override
     public void stateUpdated(Item item, State state) {
-        setState(function.calculate(members));
+        setStateWithouEvent(function.calculate(members));
+        sendGroupStateChangedEvent(item.getName(), this.state, oldGroupState);
+    }
+    
+    private void setStateWithouEvent(State state) {

```

Comment:
what is the difference of this method to setState()? The look pretty much the same to me...?